server/app/jobs/container_cleanup_job.rb

```diff
@@ -26,4 +26,28 @@ def destroy_deleted_containers
       c.destroy
     end
   end
+
+  def terminate_ghost_containers
+    Container.unscoped.where(:grid_service_id.ne => nil).each do |c|
+      if c.grid_service.nil?
+        terminate_ghost_container(c)
+      end
+    end
+  end
+
+  # @param [Docker::Container] container
+  def terminate_ghost_container(container)
+    if container.host_node
+      info "terminating #{container.to_path}"
+      terminator = Docker::ServiceTerminator.new(container.host_node)
+      terminator.request_terminate_service(
+        container.grid_service_id.to_s, container.instance_number, {lb: true}
```

Comment:
I think we'll hit similar case on the container level also. Say you have re-created a service `web` (new service from master point-of-view) and you terminate orphaned instance. That will also now remove the service instance (`web-1`) from the LB since it acts only based on names.